src/Controller/StoreApi/OrderApi/OrderApiController.php

```diff
@@ -186,10 +186,15 @@ public function giftcardStateData(SalesChannelContext $context, Request $request
      * @param Request $request
      * @return JsonResponse
      */
-    public function deleteGiftCardStateData(SalesChannelContext $context, Request $request): JsonResponse
+    public function deleteGiftCardStateData(SalesChannelContext $context, Request $request)
     {
-        $this->paymentStateDataService->deletePaymentStateDataFromContextToken($context->getToken());
-
-        return new JsonResponse(['token' => $context->getToken()]);
+        $stateData = json_decode($request->request->get('stateData', ''), true);
+        if (!empty($stateData)) {
```

Comment:
You should not use `empty` to check $stateData here. It can be not an array
```suggestion
        if (is_array($stateData) && array_key_exists('id', $stateData)) {
```

Reply:
Thank you for the insights 👍